src/features/auth/signup/SignupEmailValidation.test.tsx

```diff
@@ -0,0 +1,81 @@
+import { render, waitFor } from '@testing-library/react-native'
+import { rest } from 'msw'
+import React from 'react'
+
+import { navigate, useRoute } from '__mocks__/@react-navigation/native'
+import { ValidateEmailRequest, ValidateEmailResponse } from 'api/gen'
+import { env } from 'libs/environment'
+import { reactQueryProviderHOC } from 'tests/reactQueryProviderHOC'
+import { server } from 'tests/server'
+import { SnackBarHelperSettings } from 'ui/components/snackBar/types'
+
+import * as Auth from '../AuthContext'
+
+import { SignupEmailValidation } from './SignupEmailValidation'
+
+const mockDisplayInfosSnackBar = jest.fn()
+jest.mock('ui/components/snackBar/SnackBarContext', () => ({
+  useSnackBarContext: () => ({
+    displayInfosSnackBar: jest.fn((props: SnackBarHelperSettings) =>
+      mockDisplayInfosSnackBar(props)
+    ),
+  }),
+}))
+
+const renderPage = () => render(reactQueryProviderHOC(<SignupEmailValidation />))
+
+describe('SignupEmailValidation', () => {
+  beforeAll(() => {
+    useRoute.mockImplementation(() => ({
+      params: {
+        token: 'reerereskjlmkdlsf',
+        expiration_timestamp: 45465546445,
+      },
+    }))
+  })
+  afterEach(() => navigate.mockRestore())
+
+  it('should redirect to Home when the user is not "éligible"', async () => {
+    server.use(
+      // not eligible user call
+      rest.post<ValidateEmailRequest, ValidateEmailResponse>(
+        env.API_BASE_URL + '/native/v1/validate_email',
+        (req, res, ctx) =>
+          res.once(
+            ctx.status(200),
+            // @ts-ignore due to conflict between idCheckToken defined as string|undefined but expected null
```

Comment:
Je suis ok avec le commentaire.
Tu peux faire remonter à Victor / Xavier le besoin de rendre `idCheckToken` nullable et/ou not required ?